File: api/Entities/Admin/CallRecordItem.cs
using System.ComponentModel.DataAnnotations;

namespace api.Entities.Admin
{
    public class CallRecordItem: BaseEntity
    {
        public int CallRecordId { get; set; }
        [Required, MaxLength(3)]
        public string IncomingOutgoing {get; set;}
        public string PhoneNo {get; set;} 
        public string Email { get; set; }
        [Required]
        public DateTime DateOfContact { get; set; }
        public string Username {get; set;}
        public string GistOfDiscussions { get; set; }
        [Required, MaxLength(50)]
        public string ContactResult { get; set; }
        public string NextAction { get; set; }
        public DateTime NextActionOn { get; set; }
        public string AdvisoryBy { get; set; }
    }
}

File: client/src/app/_models/admin/callRecordItem.ts
export interface ICallRecordItem 
{
     id: number;
     incomingOutgoing: string;
     callRecordId: number;
     phoneNo: string;
     email: string;
     dateOfContact: Date;
     username: string;
     contactResult: string;
     gistOfDiscussions: string;
     nextAction: string;
     nextActionOn: Date;
     advisoryBy: string
}

export class CallRecordItem implements ICallRecordItem 
{
     id = 0;
     incomingOutgoing = "in";
     phoneNo = '';
     email='';
     callRecordId = 0;
     dateOfContact = new Date();
     username = '';
     contactResult = '';
     gistOfDiscussions = '';
     nextAction="";
     nextActionOn= new Date;
     advisoryBy="";
}